src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService2.cs

```diff
@@ -0,0 +1,201 @@
+                .ForMember(dest => dest.TimePeriodCoverage,
+                    m => m.MapFrom(r => r.ReleaseSummary.TimePeriodCoverage));
+            
+            cfg.CreateMap<CreateReleaseViewModel, ReleaseSummaryVersion>();
+            cfg.CreateMap<EditReleaseSummaryViewModel, ReleaseSummaryVersion>();
+            cfg.CreateMap<ReleaseSummary, EditReleaseSummaryViewModel>();
+        }).CreateMapper();
+
+
+        public ReleaseService2(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<Release> GetAsync(ReleaseId id)
+        {
+            return await _context.Releases.FirstOrDefaultAsync(x => x.Id == id);
+        }
+        
+        // TODO Authorisation will be required when users are introduced
+        public async Task<ReleaseViewModel> GetReleaseForIdAsync(ReleaseId id)
+        {
+            var release = await _context.Releases
+                .Where(x => x.Id == id)
+                .HydrateReleaseForReleaseViewModel()
+                .FirstOrDefaultAsync();
+            return _mapper.Map<ReleaseViewModel>(release);
+        }
+
+        // TODO Authorisation will be required when users are introduced
+        public async Task<Either<ValidationResult, ReleaseViewModel>> CreateReleaseAsync(
+            CreateReleaseViewModel createRelease)
+        {
+            return await ValidateReleaseSlugUniqueToPublication(createRelease.Slug, createRelease.PublicationId)
+                .OnSuccess(async () =>
+                {
+                    var order = OrderForNextReleaseOnPublication(createRelease.PublicationId);
+                    var content = await TemplateFromRelease(createRelease.TemplateReleaseId);
+                    var releaseSummary = _mapper.Map<ReleaseSummaryVersion>(createRelease);
+                    releaseSummary.Created = DateTime.Now;
+                    var saved = _context.Releases.Add(new Release
+                    {
+                        PublicationId = createRelease.PublicationId,
+                        ReleaseSummary = new ReleaseSummary
+                        {
+                            Versions = new List<ReleaseSummaryVersion>()
+                            {
+                                releaseSummary
+                            }
+                        },
+                        Content = content,
+                        Order = order,
+                    });
+                    
+                    await _context.SaveChangesAsync();
+                    return await GetReleaseForIdAsync(saved.Entity.Id);
+                });
+        }
+
+        // TODO Authorisation will be required when users are introduced
+        public async Task<EditReleaseSummaryViewModel> GetReleaseSummaryAsync(Guid releaseId)
+        {
+            var release = await _context.Releases
+                .Where(r => r.Id == releaseId)
+                .HydrateReleaseForReleaseViewModel()
```

Comment:
what the heck is hydrate :)

Reply:
So that's a method I've introduced case I was having 
.Include...
.Include...
.Include...
all over the place in the service methods - this puts it in one place and means I accidentally omit an include
Not sure if the name is good - but that's what it is

Reply:
- Its hydrating a object graph :)